toolchain/sem_ir/node_kind.def

```diff
@@ -7,101 +7,64 @@
 // inclusion to expand to the desired output. Macro definitions are cleaned up
 // at the end of this file.
 //
-// Exactly one of these macros should be defined before including this header:
+// This macro should be defined before including this header:
 // - CARBON_SEM_IR_NODE_KIND(Name)
 //   Invoked for each kind of semantic node.
-// - CARBON_SEM_IR_NODE_KIND_WITH_VALUE_KIND(Name, TypeFieldKind)
-//   Invoked for each kind of semantic node, along with information about
-//   whether the node produces a value, and if so, what kind of value.
-// - CARBON_SEM_IR_NODE_KIND_WITH_TERMINATOR_KIND(Name, TerminatorKind)
-//   Invoked for each kind of semantic node, along with information about
-//   whether the node is a terminator node.
-// - CARBON_SEM_IR_NODE_KIND_WITH_IR_NAME(Name, IRName)
-//   Invoked for each kind of semantic node, along with the name that is used
-//   to denote this node in textual Semantics IR.
 
-#if defined(CARBON_SEM_IR_NODE_KIND)
-#define CARBON_SEM_IR_NODE_KIND_IMPL(Name, IRName, ValueKind, TerminatorKind) \
-  CARBON_SEM_IR_NODE_KIND(Name)
-#elif defined(CARBON_SEM_IR_NODE_KIND_WITH_VALUE_KIND)
-#define CARBON_SEM_IR_NODE_KIND_IMPL(Name, IRName, ValueKind, TerminatorKind) \
-  CARBON_SEM_IR_NODE_KIND_WITH_VALUE_KIND(Name, ValueKind)
-#elif defined(CARBON_SEM_IR_NODE_KIND_WITH_TERMINATOR_KIND)
-#define CARBON_SEM_IR_NODE_KIND_IMPL(Name, IRName, ValueKind, TerminatorKind) \
-  CARBON_SEM_IR_NODE_KIND_WITH_TERMINATOR_KIND(Name, TerminatorKind)
-#elif defined(CARBON_SEM_IR_NODE_KIND_WITH_IR_NAME)
-#define CARBON_SEM_IR_NODE_KIND_IMPL(Name, IRName, ValueKind, TerminatorKind) \
-  CARBON_SEM_IR_NODE_KIND_WITH_IR_NAME(Name, IRName)
-#else
+#ifndef CARBON_SEM_IR_NODE_KIND
 #error "Must define the x-macro to use this file."
 #endif
 
 // A cross-reference between IRs.
-CARBON_SEM_IR_NODE_KIND_IMPL(CrossReference, "xref", Typed, NotTerminator)
+CARBON_SEM_IR_NODE_KIND(CrossReference)
```

Comment:
Maybe merge this into the main list? It doesn't feel special anymore, I don't think it needed to be first.

Reply:
Done.